src/plugins/kibana/public/dashboard/index.js

```diff
@@ -61,6 +65,11 @@ define(function (require) {
 
         var dash = $scope.dash = $route.current.locals.dash;
 
+        $rootScope.$broadcast('change:dashboard:darkTheme', dash.darkTheme);
```

Comment:
If you implement `chrome.addClass()` then you can just implement that here and we can just call that here rather than relying on catching an event in a higher level location. You can require the chrome module from anywhere with `require('ui/chrome')`. It doesn't need to be in public/kibana.js